### PR TITLE
Use non-SNAPSHOT hazelcast dependencies

### PR DESCRIPTION
**Pull request by JackPGreen**
To ensure future build reproducibility, use production-released dependencies over SNAPSHOT builds.

### CODE REVIEW

**Comment by JackPGreen** [comment]
Reopen

**Review by avtarraikmo — CHANGES_REQUESTED** [review]
Can we update to 5.5.0?

File: .github/hazelcast-config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+hazelcast-version: "5.3.7"
```
[the remark this answers is not in the document]
Comment (JackPGreen) [reply .github/hazelcast-config.yml:1]: > can we review and update to 5.5.0?


I can do, but unless [this PR is unblocked](https://github.com/hazelcast/hazelcast-cpp-client/issues/1222) (it's been open nearly a year) there's no point.

**Comment by JackPGreen** [comment]
Done in https://github.com/hazelcast/hazelcast-cpp-client/pull/1231

**Pull request closed without merging**